packages/cli/src/Generate.ts

```diff
@@ -248,17 +248,28 @@ This might lead to unexpected behavior.
 Please make sure they have the same version.`
             : ''
 
+        const isTS = fs.existsSync(path.resolve(cwd, 'tsconfig.json'))
```

Comment:
Hey @fiznool, thanks a lot for your contribution! I think this can work, yet I also think there will be cases where it does not:

* I think that a TS project does not necessarily have a `tsconfig.json` at the root of the package (or the file could have a different name), for instance in some monorepos it can be at the root on the monorepo only. Would it work to look if `typescript` is installed for instance?

* The JS side now changed the snippet to `require`, in ESM projects for instance you could actually want the `import`, so how do we detect ESM projects?

Reply:
Thanks for your review. On the two points:

1. I'm not quite sure how you'd detect the presence of typescript in a non-hacky way. Do you have any thoughts on how to do this? I can think of two possibilities: using `require.resolve('typescript)` to try and load typescript, or to run the `npm ls typescript--depth=0` command. Neither feels like an elegant solution!
2. I did originally consider the ES Modules possibility, however on consultation with the [docs](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client), it seems that the code snippets for JS projects still use CommonJS. I therefore decided to maintain parity with the documentation. Does that make sense?

Reply:
I think `require.resolve` is a good solution, albeit not pretty. It may be more eager than looking for `tsconfig` which in some cases could also technically be wrong (eg. JS package in monorepo that partially uses TS), in that case we would get the current behavior and maybe that's fine.

On your second point, yes that makes sense. And you reminded me that we actually had some reports that ESM imports need to work with a workaround at the moment, so I think this is fine. Also, ESM detection should be straightforward.

Reply:
Just pushed a change to use `require.resolve` for detecting if TypeScript is installed, as discussed.

Reply:
The problem we have now is that we can’t really add a test for this, since in the test harness, typescript will always be resolved. I guess we’ll have to accept that this can’t be tested? In the same way, for example, that the `isDeno` branches also cannot be tested here.

Reply:
In the CLI tests (`prisma/packages/prisma/src/__tests__`) we have fixtures that could help you simulate a project with and without TypeScript. That said, I don't think we install dependencies in there, so that may be a problem. Another way would be to create a fake TypeScript module in the node_modules of the one fixture and not in the other.

Alternatively, you can add a test in `prisma/packages/client/tests/e2e`. We also have a CLI for running these, via `pnpm run test:e2e example --verbose`, so you could duplicate the example folder in there. These are more costly to write and to run.

Reply:
I think the main problem here is that `require.resolve` is going to recursively walk the tree until it finds TypeScript, which will be installed at the root of the tree when the tests are run. So I don't see an easy way to mock this, unless I mock the `canResolveTypeScript` function? Alternatively, I'm not sure if this is really worth adding test coverage for.

Reply:
`test:e2e` should be run in a completely isolated test in a docker container, while the fixtures would be run in a folder in `/tmp`